### PR TITLE
fix(vars): match Python round semantics

### PR DESCRIPTION
**Pull request by anxkhn**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls) for the desired changed?

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Description

Compiled `NumberVar` rounding previously used JavaScript `Math.round` and
`Number.prototype.toFixed`. These APIs differ from Python on half ties, and
`toFixed` rejects negative `ndigits` values.

This change routes compiled rounding through a frontend helper that decomposes
IEEE 754 values into exact rational numbers and applies round-half-even. It adds
support for negative precision while preserving the existing static result types.

### Testing

- `uv run pytest tests/units/test_var.py -q` (223 passed)
- `uv run pytest tests/units --cov --no-cov-on-fail --cov-report=` (6664 passed, 17 skipped, 78.09% coverage)
- `uv run ruff check .`
- `uv run ruff format --check .`
- `uv run pyright reflex tests` (0 errors)
- `uv run pre-commit run --files packages/reflex-base/src/reflex_base/.templates/web/utils/helpers/round.js packages/reflex-base/src/reflex_base/vars/number.py packages/reflex-base/news/6757.bugfix.md tests/units/test_var.py`
- `node --check packages/reflex-base/src/reflex_base/.templates/web/utils/helpers/round.js`
- `uv run towncrier check --config pyproject.toml --dir packages/reflex-base --compare-with origin/main`

The regression test executes the generated JavaScript in Node and covers positive
and negative half ties, negative precision, binary-float edge cases, extreme
precision, signed zero, non-finite values, helper imports, and result types.

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: packages/reflex-base/src/reflex_base/.templates/web/utils/helpers/round.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const doubledRemainder = (numerator % denominator) * 2n;
+  if (
+    doubledRemainder > denominator ||
+    (doubledRemainder === denominator && rounded % 2n === 1n)
+  ) {
+    rounded += 1n;
+  }
+
+  const sign = value < 0 || Object.is(value, -0) ? "-" : "";
+  const result = Number(`${sign}${rounded}e${-ndigits}`);
```
[the remark this answers is not in the document]
Comment (anxkhn) [reply packages/reflex-base/src/reflex_base/.templates/web/utils/helpers/round.js:44]: addressed in e2271f18: non-finite conversion results now raise rangeerror, with the max-value case covered by the regression test.

File: packages/reflex-base/src/reflex_base/.templates/web/utils/helpers/round.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * Round a number using Python's round-half-even semantics.
+ * Decompose the IEEE 754 value into a rational so ties are compared exactly.
+ *
+ * @param {number} value The value to round.
+ * @param {number} ndigits The decimal precision.
+ * @returns {number} The rounded value.
+ */
+export default function pyRound(value, ndigits) {
+  if (!Number.isFinite(value) || ndigits > 323) return value;
+  if (ndigits < -308) return value * 0;
```
[the remark this answers is not in the document]
Comment (anxkhn) [reply packages/reflex-base/src/reflex_base/.templates/web/utils/helpers/round.js:14]: addressed in e2271f18: fractional ndigits now raise typeerror before bigint conversion, matching python input validation.

File: tests/units/test_var.py [hunk 3]
```diff
@@ -1049,6 +1056,51 @@ def test_all_number_operations():
...
+    )
+
+    result = subprocess.run(
+        [
+            "node",
+            "--input-type=module",
+            "--eval",
+            f"{ROUND_HELPER.read_text(encoding='utf-8')}\nconsole.log({rounded!s})",
+        ],
+        check=True,
```
[the remark this answers is not in the document]
Comment (anxkhn) [reply tests/units/test_var.py:1098]: the repository test ci provides node for frontend compilation and helper tests, so this dependency is covered by ci rather than skipped. the targeted cases also pass locally under node.

**Comment by FarhanAliRaza** [comment]
Thank you for your work. 
This is a slow solution of problem that is not actually a real problem.
